crates/sui-core/src/execution_cache/memory_cache.rs

```diff
@@ -0,0 +1,1143 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! MemoryCache is a cache for the transaction execution which delays writes to the database until
+//! transaction results are certified (i.e. they appear in a certified checkpoint, or an effects cert
```

Comment:
One basic question related to data quarantine: are effects quarantined as well, before they are certified in a checkpoint?

Reply:
yes, all transaction outputs will be quarantined until proven to be agreed upon by a quorum